core/src/main/java/org/infinispan/marshall/exts/OptionalExternalizer.java

```diff
@@ -0,0 +1,38 @@
+package org.infinispan.marshall.exts;
+
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Optional;
+import java.util.Set;
+
+public class OptionalExternalizer extends AbstractExternalizer<Optional> {
```

Comment:
I wonder if we really need this externalizer.  It seems most operations would be fine returning a null and then we do the Optional wrapping locally?  If not do we want to worry about the various primitive optionals as well (OptionalInt etc.)?